src/components/StarRating/src/StarRating.vue

```diff
@@ -0,0 +1,139 @@
+<template>
+	<div :style="starComputedStyles">
+		<star
+			v-for="star in stars"
+			:key="star"
+			:fill="fillForStar(star)"
+			:class="$s.Star"
+			@hover="hoverStar(star)"
+			@unhover="unhoverStar(star)"
+			@click="clickStar(star)"
+		/>
+	</div>
+</template>
+
+<script>
+import Star from './Star.vue';
+
+const MIN_RATING = 0;
+const MAX_RATING = 5;
+const HALF_RATING = 0.5;
+// eslint-disable-next-line no-magic-numbers
+const stars = [1, 2, 3, 4, 5];
+
+export default {
+	name: 'StarRating',
+
+	components: {
+		Star,
+	},
+
+	props: {
+		/**
+		 * Rating value from 0-5, determines fill state of stars
+		 */
+		rating: {
+			type: Number,
+			default: 0,
+			validate: (rating) => rating >= MIN_RATING && rating <= MAX_RATING,
+		},
+
+		/**
+		 * Size of rating component
+		 */
+		size: {
+			type: String,
+			default: 'medium',
+			validate: (size) => ['small', 'medium', 'large'].includes(size),
+		},
+
+		/**
+		 * Determines whether to bubble up click/hover events and show pointer cursor
+		 */
+		selectable: {
+			type: Boolean,
+			default: false,
+		},
+	},
+
+	data() {
+		return {
+			stars,
+			styles: {
```

Comment:
If these are not mutable, they could just be defined as a constant with string values?

```
const SIZE_STYLES: {
    small: {
        height: '16px',
        spacing: '1px',
    },
    medium: {
        height: '24px',
        spacing: '2px',
    },
    large: {
        height: '32px',
        spacing: '4px',
    },
},
```

```
const styles = SIZE_STYLES[this.size];

return {
    '--star-height': styles.height,
    '--star-spacing': styles.spacing,
    '--star-cursor': this.selectable ? 'pointer' : 'default',
};
```